hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientExecutionServiceImpl.java

```diff
@@ -149,19 +150,11 @@ public void run() {
     }
 
     @Override
-    public ScheduledFuture<?> scheduleAtFixedRate(final Runnable command, long initialDelay, long period, TimeUnit unit) {
+    public ScheduledFuture<?> scheduleWithRepetition(final Runnable command, long initialDelay, long period, TimeUnit unit) {
+        final Runnable decoratedCommand = new SkipOnConcurrentExecutionDecorator(command);
         return scheduledExecutor.scheduleAtFixedRate(new Runnable() {
             public void run() {
-                executeInternalSafely(command, failureLoggingExecutionCallback);
-            }
-        }, initialDelay, period, unit);
-    }
-
-    @Override
-    public ScheduledFuture<?> scheduleWithFixedDelay(final Runnable command, long initialDelay, long period, TimeUnit unit) {
-        return scheduledExecutor.scheduleWithFixedDelay(new Runnable() {
-            public void run() {
-                executeInternalSafely(command, failureLoggingExecutionCallback);
+                executeInternalSafely(decoratedCommand, failureLoggingExecutionCallback);
```

Comment:
I double checked it and it's not really an issue: the `ClientExecutionService` does not accept custom callback for `scheduleWithRepetition`. 

The service does use the callback internally, but only to logs errors -> not a problem.